frontend/src/components/transactions/Transactions.tsx

```diff
@@ -15,47 +16,59 @@ export interface State {
   transactions: T.Transaction[] | null;
 }
 
-export default class extends React.PureComponent<Props, State> {
+export default class extends React.Component<Props, State> {
   static defaultProps = {
     reversed: false,
-    limit: 50
+    limit: 15
   };
 
   state: State = {
     transactions: null
   };
 
   _transactionsApi: TransactionsApi | null;
+  timer: any;
 
   constructor(props: Props) {
     super(props);
 
     // TODO: Design ExplorerApi to handle server-side rendering gracefully.
     this._transactionsApi = null;
+    this.timer = null;
   }
 
   componentDidMount() {
-    this.fetchTransactions();
+    this.timer = setTimeout(this.regularFetchInfo, 0);
+    this.regularFetchInfo();
   }
 
-  componentDidUpdate(prevProps: Props) {
-    if (this.props !== prevProps) {
-      this.fetchTransactions();
-    }
+  componentWillUnmount() {
+    clearTimeout(this.timer);
+    this.timer = null;
   }
 
-  render() {
-    const { transactions } = this.state;
-    if (transactions === null) {
-      return null;
+  regularFetchInfo = async () => {
+    if (this.state.transactions === null) {
+      // first time to fetch data
+      await this.fetchTransactions();
+    } else {
+      // later time
+      await this.fetchLatestTransactions();
```

Comment:
We don't need this special case at all. It also breaks pages like Account Details: https://near-explorer-frontend-pr-133.onrender.com/accounts/test-seed-phrase (wait 10 seconds and see what happens)

Reply:
no problem, deleted that